zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/MultiResourceDeploymentTest.java

```diff
@@ -0,0 +1,341 @@
+      assertThat(secondDeployment.getProcessesMetadata())
+          .singleElement()
+          .satisfies(
+              metadata ->
+                  Assertions.assertThat(metadata)
+                      .hasVersion(1)
+                      .isDuplicate()
+                      .extracting(
+                          ProcessMetadataValue::getProcessDefinitionKey,
+                          InstanceOfAssertFactories.LONG)
+                      .isEqualTo(processV1.getProcessDefinitionKey()));
+      assertThat(secondDeployment.getDecisionRequirementsMetadata())
+          .singleElement()
+          .satisfies(
+              metadata ->
+                  Assertions.assertThat(metadata)
+                      .hasDecisionRequirementsVersion(1)
+                      .isDuplicate()
+                      .extracting(
+                          DecisionRequirementsMetadataValue::getDecisionRequirementsKey,
+                          InstanceOfAssertFactories.LONG)
+                      .isEqualTo(drgV1.getDecisionRequirementsKey()));
+      assertThat(secondDeployment.getDecisionsMetadata())
+          .singleElement()
+          .satisfies(
+              metadata ->
+                  Assertions.assertThat(metadata)
+                      .hasVersion(1)
+                      .isDuplicate()
+                      .extracting(
+                          DecisionRecordValue::getDecisionKey,
+                          DecisionRecordValue::getDecisionRequirementsKey)
+                      .containsExactly(
+                          decisionV1.getDecisionKey(), drgV1.getDecisionRequirementsKey()));
+      assertThat(secondDeployment.getFormMetadata())
+          .singleElement()
+          .satisfies(
+              metadata ->
+                  Assertions.assertThat(metadata)
+                      .hasVersion(1)
+                      .isDuplicate()
+                      .extracting(FormMetadataValue::getFormKey)
+                      .isEqualTo(formV1.getFormKey()));
+
+      assertThat(RecordingExporter.processRecords().withIntent(ProcessIntent.CREATED).limit(2))
+          .map(record -> record.getValue().getProcessDefinitionKey())
+          .containsExactly(processV1.getProcessDefinitionKey());
+      assertThat(
+              RecordingExporter.decisionRequirementsRecords()
+                  .withIntent(DecisionRequirementsIntent.CREATED)
+                  .limit(2))
+          .map(record -> record.getValue().getDecisionRequirementsKey())
+          .containsExactly(drgV1.getDecisionRequirementsKey());
+      assertThat(RecordingExporter.decisionRecords().withIntent(DecisionIntent.CREATED).limit(2))
+          .map(record -> record.getValue().getDecisionKey())
+          .containsExactly(decisionV1.getDecisionKey());
+      assertThat(RecordingExporter.formRecords().withIntent(FormIntent.CREATED).limit(2))
+          .map(record -> record.getValue().getFormKey())
+          .containsExactly(formV1.getFormKey());
```

Comment:
Executing this section of the code takes more than 20 seconds on my machine 🤷🏻. I'm not sure if there's a way to make it faster?

Reply:
Hi Jan, we should look at the ways to improve execution time, myself and ZPA team will assist on that.